### PR TITLE
Use `:` for time and showtime msg formatting

### PR DESCRIPTION
**Pull request by IanButterworth**
As suggested in https://github.com/JuliaLang/julia/pull/42431#issuecomment-953239795

Changes `->` to `:` and removes the left-padding on the time report when a description is provided
```
julia> @showtime sleep(1)
sleep(1): 1.002159 seconds (286 allocations: 15.273 KiB)

julia> @time "description" sleep(1)
description: 1.002132 seconds (5 allocations: 144 bytes)

julia> @time sleep(1)
  1.001287 seconds (5 allocations: 144 bytes)
```

### CODE REVIEW

File: base/timing.jl [hunk 1]
```diff
@@ -323,8 +324,9 @@ macro timev(msg, ex)
             compile_elapsedtime = cumulative_compile_time_ns_after() - compile_elapsedtime)
         )
         local diff = GC_Diff(gc_num(), stats)
-        isnothing($(esc(msg))) || print($(esc(msg)), " ->")
-        timev_print(elapsedtime, diff, compile_elapsedtime)
+        local has_msg = !isnothing($(esc(msg)))
+        has_msg && print($(esc(msg)), " : ")
```
Comment (fredrikekre) [inline base/timing.jl:328]: I think it looks better without the first space
```suggestion
        has_msg && print($(esc(msg)), ": ")
```

File: base/timing.jl [hunk 2]
```diff
@@ -243,8 +243,9 @@ macro time(msg, ex)
             compile_elapsedtime = cumulative_compile_time_ns_after() - compile_elapsedtime)
         )
         local diff = GC_Diff(gc_num(), stats)
-        isnothing($(esc(msg))) || print($(esc(msg)), " ->")
-        time_print(elapsedtime, diff.allocd, diff.total_time, gc_alloc_count(diff), compile_elapsedtime, true)
+        local has_msg = !isnothing($(esc(msg)))
```
Comment (JeffBezanson) [inline base/timing.jl:246]: This should use a temporary variable to avoid evaluating `msg` twice.

File: base/timing.jl [hunk 3]
```diff
@@ -243,8 +243,9 @@ macro time(msg, ex)
...
+        has_msg && print($(esc(msg)), " : ")
```
Comment (JeffBezanson) [inline base/timing.jl:247]: Remove the extra space here too?

**Comment by IanButterworth** [comment]
FreeBSD error is HTTPBingo flakiness

**Pull request merged**